src/AppInstallerCLICore/ExecutionReporter.cpp

```diff
@@ -155,6 +155,27 @@ namespace AppInstaller::CLI::Execution
         }
     }
 
+    std::filesystem::path Reporter::PromptForPath(Resource::LocString message, Level level)

```

Comment:
Re 1. I'm adding a Context flag to set non-interactive mode. The flag is set when constructing a COMContext and can be added any other place we need it. Then each prompt has to handle the flag however it needs to (e.g. the one for abort terminal does nothing, and the one for agreements throws an error).

Re 2 & 3. @denelon, what do you think about adding an argument and setting to suppress prompts (agreements, missing install location, warning about installer aborting the terminal)?

Reply:
RE 1: Agreed that each location will have to decide what the non-interactive default is, and then some or all of them will need to have additional arguments to select the non-default behavior.

RE 2&3: I don't think this is an option, but a requirement.

Reply:
Also, you cannot have the agreement throwing an error in the COM case without:

1. Exposing the agreements through COM
2. Enabling the ability to accept the agreements through COM
3. Giving a grace period for callers to start using this pattern

Reply:
Added a context flag for COM, a CLI arg and a setting to indicate non-interactive.

I'm not sure what to do for a proper handling of the agreements via COM, but this change does not leave us worse off than we currently are. Currently, when a COM context is created, a flag is set that signals that the agreements were accepted by the caller, so it will not throw.